File: ScorpioConversion/src/Reader/IReader.cs
using System;

public interface IReader : IBase {
    public void ReadHead(string fileName);
    public bool ReadBool();
    public sbyte ReadInt8();
    public byte ReadUInt8();
    public short ReadInt16();
    public ushort ReadUInt16();
    public int ReadInt32();
    public uint ReadUInt32();
    public long ReadInt64();
    public ulong ReadUInt64();
    public float ReadFloat();
    public double ReadDouble();
    public string ReadString();
    public DateTime ReadDateTime();
    public byte[] ReadBytes();
}

File: Scorpio.Conversion.Runtime/Go/ScorpioConversionRuntime/IReader.go
package ScorpioConversionRuntime

import "time"

type IReader interface {
	ReadBool() bool
	ReadInt8() int8
	ReadUInt8() uint8
	ReadInt16() int16
	ReadUInt16() uint16
	ReadInt32() int32
	ReadUInt32() uint32
	ReadInt64() int64
	ReadUInt64() uint64
	ReadFloat() float32
	ReadDouble() float64
	ReadString() string
	ReadDateTime() time.Time
	Close()
}
